.github/workflows/golangci-lint.yml

```diff
@@ -0,0 +1,23 @@
+name: golint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+  pull_request:
+jobs:
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: latest
+          args: --new-from-rev=HEAD~
+          level: warning # GitHub Status Check won't become failure with this level.
```

Comment:
why do you use the warning level? We do want that the action fails if it doesn't pass 
please add a new line at the end.

Reply:
I've copied this section from the file you've uploaded on your PR